filament/backend/src/metal/MetalDriver.mm

```diff
@@ -706,16 +706,26 @@
     scheduleDestroy(std::move(data));
 }
 
-void MetalDriver::setExternalIndexBuffer(Handle<HwIndexBuffer> ibh, void* externalBuffer) {
+void MetalDriver::setupExternalResource(intptr_t externalResource) {
+    CFTypeRef bufferRef = (CFTypeRef) externalResource;
+    CFRetain(bufferRef);
+}
+
+void MetalDriver::cancelExternalResource(intptr_t externalResource) {
```

Comment:
`setupExternalImage(void*)` and `cancelExternalImage(void*)` functions can now be removed from the driver API, as those are superceded by these. The existing comment there could be adapted, mentioning that here we take ownership of the passed in buffer/texture, and that objects of type `CVPixelBufferRef`, `id<MTLBuffer>`, `id<MTLTexture>` are supported/expected.

Reply:
I removed both of them in the commit: fbd7516
Apparently `setupExternalResource` for `OpenGLDriver` in case of `PlatformCocoaTouchGLImpl` is _not_ a no-op, so I implemented a the retaining there as well.